internal/rbd/rbd_util.go

```diff
@@ -793,14 +809,14 @@ func (rv *rbdVolume) checkImageChainHasFeature(ctx context.Context, feature uint
 			if errors.Is(err, ErrImageNotFound) {
 				return false, nil
 			}
-			util.ErrorLog(ctx, "failed to get image info for %s: %s", vol.String(), err)
+			util.ErrorLog(ctx, "failed to get parent image info for %s: %s", currVol.String(), err)
 			return false, err
 		}
-		if f := vol.hasFeature(feature); f {
+		defer parentVol.Destroy()
```

Comment:
defer in for loop as gotchas https://blog.learngoprogramming.com/gotchas-of-defer-in-go-1-8d070894cb01